### PR TITLE
feat(minor-multisig-prover): remove external gateway encoding v1

### PR DESCRIPTION
**Pull request by haiyizxx**
## Description

- remove deprecated types
- remove abi encoding
- remove bsc encoding, will add the new bsc encoding when the new external Sui Gateway is ready

## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: contracts/multisig-prover/src/encoding/abi2/mod.rs [hunk 1]
```diff
@@ -143,14 +143,34 @@ fn evm_address(pub_key: &MultisigPublicKey) -> Result<Address, ContractError> {
     }
 }
 
+pub fn make_operators(worker_set: WorkerSet) -> Operators {
```
Comment (haiyizxx) [inline contracts/multisig-prover/src/encoding/abi2/mod.rs:146]: Took from [abi::make_operators](https://github.com/axelarnetwork/axelar-amplifier/blob/multisig-prover-v0.2.0/contracts/multisig-prover/src/encoding/abi.rs#L133) and add a unit test

File: contracts/multisig-prover/src/encoding/abi2/mod.rs
Comment (milapsheth) [inline contracts/multisig-prover/src/encoding/abi2/mod.rs:1]: rename the folder to `abi`?

**Review by milapsheth — APPROVED** [review]

File: contracts/multisig-prover/src/encoding/abi2/mod.rs
Comment (haiyizxx) [reply contracts/multisig-prover/src/encoding/abi2/mod.rs:1]: plan to do in a separate pr

File: contracts/multisig-prover/src/encoding/abi2/mod.rs (hunk 1 above)
Comment (milapsheth) [inline contracts/multisig-prover/src/encoding/abi2/mod.rs:146]: implement TryFrom for this cast?

**Pull request merged**